xarray/core/utils.py

```diff
@@ -62,8 +62,11 @@ def wrapper(*args, **kwargs):
 def _maybe_cast_to_cftimeindex(index: pd.Index) -> pd.Index:
     from ..coding.cftimeindex import CFTimeIndex
 
-    if index.dtype == 'O':
+    if len(index) > 0 and index.dtype == 'O':
         try:
+            import cftime
+            if not isinstance(index[0], cftime.datetime):
```

Comment:
We have a check similar to this that gets called in the `CFTimeIndex` constructor, so I think we may not need this check here:
https://github.com/pydata/xarray/blob/5343ccc4bef77020a8181c9da0fdee6bcae35b5f/xarray/coding/cftimeindex.py#L188-L193

Reply:
It looks like I was really tired... The name of the function was `assert_all_valid_date_type` so I somehow assume this would test all elements are valid not just the first :)